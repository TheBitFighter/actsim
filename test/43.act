defproc recv(chan?(int) X)
{
  int i;
  chp {
     i := 0;
    *[ i < 100 -> X?; i := i + 1 ]
   }
}

defproc mytest(chan!(int) A, B)
{
  chp {
    *[ A!0,B!1 ]
   }
}

defproc test()
{
  chan(int) A, B, C, D;
  int a, b, c, d;

  recv R0(A),R1(B),R2(C),R3(D);

  chp { 
    *[ A!0,B!1 ] , *[ C!1,D!2 ]
  }
}

test t;
