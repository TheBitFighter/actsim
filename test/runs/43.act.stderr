WARNING: test<>: substituting chp model (requested prs, not found)
WARNING: recv<>: substituting chp model (requested prs, not found)
WARNING: recv<>: substituting chp model (requested prs, not found)
WARNING: recv<>: substituting chp model (requested prs, not found)
WARNING: recv<>: substituting chp model (requested prs, not found)
